src/core/commit.rs

```diff
@@ -203,6 +203,14 @@ impl<J: Jet> CommitNode<J> {
         self.arrow.target_ty()
     }
 
+    pub fn iter(&self) -> PostOrderIter<RefWrapper<J>> {
+        RefWrapper(self).iter_post_order()
+    }
+
+    pub fn size(&self) -> usize {
+        self.iter().count()
```

Comment:
In b3f7a5e090101ed405fcbc53de1021ccaed18bf1:

Again, pre-sharing or post-sharing? Probably we should add a FIXME comment saying to something more efficient than counting every single node like this.

Also the function should be called `len` rather than `size`, no?

Reply:
A graph doesn't have a length, but since we are calling these graphs "programs" a length is probably more intuitive.

Reply:
Iterators, hashmaps, btrees, etc., don't have lengths etiher, but in Rust it is conventional to use `len` to refer to the size operator, and `size` only to refer to encoding sizes.